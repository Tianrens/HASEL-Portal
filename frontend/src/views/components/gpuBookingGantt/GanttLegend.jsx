import React from 'react';
import colours from '../../../assets/_colours.module.scss';
import styles from './GanttLegend.module.scss';

export default function GanttLegend({ hasCurrentBooking }) {
    const categories = [
        { description: 'Your Bookings', colour: colours.secondaryBlue },
        { description: 'Other Bookings', colour: colours.gray },
    ];

    if (hasCurrentBooking) {
        categories.push({ description: 'Current Booking', colour: colours.green });
    }

    return (
        <div className={styles.wrapper}>
            <div className={styles.title}>Legend</div>
            <div className={styles.categoriesWrapper}>
                {categories.map((category) => (
                    <div key={category.description} className={styles.category}>
                        <div className={styles.dot} style={{ backgroundColor: category.colour }} />
                        {category.description}
                    </div>
                ))}
            </div>
        </div>
    );
}
