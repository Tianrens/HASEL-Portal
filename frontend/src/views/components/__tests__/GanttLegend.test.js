import React from 'react';
import renderer from 'react-test-renderer';
import { StylesProvider } from '@material-ui/core';
import GanttLegend from '../gpuBookingGantt/GanttLegend';

test('GanttLegend renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <GanttLegend hasCurrentBooking />
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
