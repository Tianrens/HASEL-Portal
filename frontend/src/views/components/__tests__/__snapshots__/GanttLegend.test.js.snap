// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GanttLegend renders properly 1`] = `
<div
  className="wrapper"
>
  <div
    className="title"
  >
    Legend
  </div>
  <div
    className="categoriesWrapper"
  >
    <div
      className="category"
    >
      <div
        className="dot"
        style={
          Object {
            "backgroundColor": "secondaryBlue",
          }
        }
      />
      Your Bookings
    </div>
    <div
      className="category"
    >
      <div
        className="dot"
        style={
          Object {
            "backgroundColor": "gray",
          }
        }
      />
      Other Bookings
    </div>
    <div
      className="category"
    >
      <div
        className="dot"
        style={
          Object {
            "backgroundColor": "green",
          }
        }
      />
      Current Booking
    </div>
  </div>
</div>
`;
